src/auth0/cache/mod.rs

```diff
@@ -29,8 +29,17 @@ pub trait Cache: Send + Sync + std::fmt::Debug {
     async fn get_token(&self, client_id: &str, aud: &str) -> Result<Option<Token>, CacheError>;
 
     async fn put_token(&self, client_id: &str, aud: &str, token: &Token) -> Result<(), CacheError>;
-}
 
-pub(in crate::auth0::cache) fn token_key(caller: &str, audience: &str) -> String {
-    format!("{}:{}:{}:{}", TOKEN_PREFIX, caller, TOKEN_VERSION, audience)
+    fn service_name(&self) -> &str;
+
+    fn token_key(&self, caller: &str, audience: &str) -> String {
```

Comment:
Can we document that tokens should be prefixed by a service name so nobody changes this in the future?

Reply:
Added a comment, not sure if you had something more in mind